packages/dialog/test/lion-dialog.test.js

```diff
@@ -30,5 +29,30 @@ describe('lion-dialog', () => {
 
       expect(el.opened).to.be.true;
     });
+
+    it('supports nested overlays', async () => {
+      const el = await fixture(html`
+        <lion-dialog>
+          <div slot="content">
+            open nested overlay:
+            <lion-dialog>
+              <div slot="content">
+                Nested content
+              </div>
+              <button slot="invoker">nested invoker button</button>
+            </lion-dialog>
+          </div>
+          <button slot="invoker">invoker button</button>
+        </lion-dialog>
+      `);
+
+      el._overlayInvokerNode.click();
+      expect(el.opened).to.be.true;
+
+      const wrapperNode = Array.from(document.querySelector('.global-overlays').children)[1];
```

Comment:
I had to do it this way, because if you create a node reference for the inner dialog, it renders fully before being moved. But in a real scenario where you create this template, the inner dialog doesn't get fully rendered, because it gets moved earlier. There's a timing difference, and for the node reference you don't get a failing test, whereas for a nested template like this, you do get this failing test.